File: Irrgarten-Java/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;


/**
 *
 * @author dpant
 */
public class Shield extends CombatElement  {
    
    Shield(float protection_, int uses_){
       super(protection_,uses_);
    }
    
    public float protect(){
        return produceEffect(); 
    }
    
    @Override
    public String toString(){
       return "S"+super.toString();  
    }

}

File: Irrgarten-ruby/Irrgarten/shield.rb
# frozen_string_literal: true

require_relative 'dice'
module Irrgarten

class Shield < CombatElement

  def initialize(p,u)
    super(p,u)
  end

  def protect
     produce_effect
  end

  def to_string
    "S[#{super.to_string}"
  end
  public_class_method :new
end
end
